rackcity/views/pdu_views.py

```diff
@@ -0,0 +1,74 @@
+# from django.http import HttpResponse, JsonResponse
+from django.http import JsonResponse
+from rackcity.models import (
+    Rack,
+    Asset
+)
+from rackcity.api.serializers import (
+    serialize_power_connections,
+)
+# from rackcity.utils.log_utils import (
+#     log_action,
+#     log_bulk_import,
+#     log_delete,
+#     Action,
+#     ElementType,
+# )
+from rest_framework.decorators import permission_classes, api_view
+from rest_framework.permissions import IsAuthenticated, IsAdminUser
+# from rest_framework.parsers import JSONParser
+from http import HTTPStatus
+from rackcity.views.rackcity_utils import (
+    validate_asset_location,
+    validate_location_modification,
+    no_infile_location_conflicts,
+    records_are_identical,
+    get_sort_arguments,
+    get_filter_arguments,
+)
+import re
+import requests
+
+pdu_url = 'http://hyposoft-mgt.colab.duke.edu:8000/pdu.php?pdu=hpdu-rtp1-' # Need to specify rack + side in request, e.g. for A1 left, use A01L
+
+
+@api_view(['GET'])
+@permission_classes([IsAuthenticated])
+def power_status(request, id):
+    """
+    Get status of all power ports for an asset in
+    network controlled PDU datacenter.
+    """
+    try:
+        asset = Asset.objects.get(id=id)
+    except Asset.DoesNotExist:
+        failure_message = "No model exists with id=" + str(id)
+        return JsonResponse(
+            {"failure_message": failure_message},
+            status=HTTPStatus.BAD_REQUEST
+        )
+    port_info = serialize_power_connections(asset)
+
+    # get string parameter representing rack number (i.e. A01<L/R>)
+    rack_str = str(asset.rack.row_letter)
+    if (asset.rack.rack_num / 10 < 1):
+        rack_str = rack_str + "0"
+    rack_str = rack_str + str(asset.rack.rack_num)
```

Comment:
I think you can do like a %s thing to format the number into the string but this is fine